src/System.Private.ServiceModel/src/System/ServiceModel/Channels/MessageContent.cs

```diff
@@ -21,23 +22,21 @@ internal abstract class MessageContent : HttpContent
         protected MessageEncoder _messageEncoder;
         protected Stream _stream = null;
         private bool _disposed;
+        protected TaskCompletionSource<bool> _writeCompletedTcs;
 
         public MessageContent(Message message, MessageEncoder messageEncoder)
         {
             _message = message;
             _messageEncoder = messageEncoder;
+            _writeCompletedTcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
 
             SetContentType(_messageEncoder.ContentType);
             PrepareContentHeaders();
         }
 
-        public Message Message
-        {
-            get
-            {
-                return _message;
-            }
-        }
+        public Message Message { get { return _message; } }
+
+        internal  Task WriteCompletionTask { get { return _writeCompletedTcs?.Task; } }
```

Comment:
This is a remnant from an earlier version of the fix where I was lazily creating the TCS. I realized there was too much possibility for race conditions so abandoned that approach. I'll remove the null check.